web/src/lightbox.js

```diff
@@ -241,8 +255,27 @@ function display_video(payload) {
     render_lightbox_list_images(payload.preview);
 
     $(
-        "#lightbox_overlay .image-preview, .image-description, .download, .lightbox-zoom-reset",
+        "#lightbox_overlay .image-preview, .image-description, .download, .lightbox-zoom-reset, .video-player",
     ).hide();
+    $(".player-container").show();
+
+    if (payload.type === "inline-video") {
+        $(".player-container").hide();
+        $(".video-player, .image-description").show();
+        const $video = $("<video>");
+        $video.attr("src", payload.source);
+        $video.attr("controls", true);
+        $(".video-player").html($video);
+        $(".image-actions .open").attr("href", payload.source);
```

Comment:
The "open" button does not seem to open a video in a new tab like it does for images; instead downloading it; might be a bug?

Reply:
This seems to need a backend change to have file response header have a
`Content-Disposition: inline` in `serve_file`. Would it be cleaner to have a new  API endpoint for this?

Also, I don't think we can modify header of external URLs.

I think I was testing this feature with a link like [this](https://file-examples.com/storage/feb57324aa650f65fa00875/2020/03/file_example_WEBM_480_900KB.webm) which seems to be working correctly.

Reply:
That does seem right. @andersk @alexmv is it correct to add the video file types to `INLINE_MIME_TYPES`?

```
INLINE_MIME_TYPES = [                                                                                   
    "application/pdf",                                                                                  
    "image/gif",                                                                                        
    "image/jpeg",                                                                                       
    "image/png",                                                                                        
    "image/webp",                                                                                       
    # To avoid cross-site scripting attacks, DO NOT add types such                                      
    # as application/xhtml+xml, application/x-shockwave-flash,                                          
    # image/svg+xml, text/html, or text/xml.                                                            
]                                                                                                       
```

Reply:
Yes, adding to that seems right to me.  I think we'll need a migration to adjust the content-disposition in S3, however.

Reply:
OK, we can do that via finishing https://github.com/zulip/zulip/pull/9059 as a follow-up to this PR.